test/e2e/common/node/pod_resize.go

```diff
@@ -71,1205 +71,543 @@ func offsetMemory(index int64, value string) string {
-					InitCtr:       true,
-					RestartPolicy: v1.ContainerRestartPolicyAlways,
-				},
-			},
-		},
-		{
-			name: "Guaranteed QoS pod, one container, one restartable init container - decrease init container CPU",
-			containers: []podresize.ResizableContainerInfo{
-				{
-					Name:      "c1",
-					Resources: &cgroups.ContainerResources{CPUReq: originalCPU, CPULim: originalCPU, MemReq: originalMem, MemLim: originalMem},
-				},
-				{
-					Name:          "c1-init",
-					Resources:     &cgroups.ContainerResources{CPUReq: originalCPU, CPULim: originalCPU, MemReq: originalMem, MemLim: originalMem},
-					InitCtr:       true,
-					RestartPolicy: v1.ContainerRestartPolicyAlways,
-				},
-			},
-			expected: []podresize.ResizableContainerInfo{
-				{
-					Name:      "c1",
-					Resources: &cgroups.ContainerResources{CPUReq: originalCPU, CPULim: originalCPU, MemReq: originalMem, MemLim: originalMem},
-				},
-				{
-					Name:          "c1-init",
-					Resources:     &cgroups.ContainerResources{CPUReq: reducedCPU, CPULim: reducedCPU, MemReq: originalMem, MemLim: originalMem},
-					InitCtr:       true,
-					RestartPolicy: v1.ContainerRestartPolicyAlways,
-				},
-			},
-		},
-		{
-			name: "Burstable QoS pod, one container, one restartable init container - increase init container CPU & memory",
-			containers: []podresize.ResizableContainerInfo{
-				{
-					Name:      "c1",
-					Resources: &cgroups.ContainerResources{CPUReq: originalCPU, CPULim: originalCPULimit, MemReq: originalMem, MemLim: originalMemLimit},
-				},
-				{
-					Name:          "c1-init",
-					Resources:     &cgroups.ContainerResources{CPUReq: originalCPU, CPULim: originalCPULimit, MemReq: originalMem, MemLim: originalMemLimit},
-					InitCtr:       true,
-					RestartPolicy: v1.ContainerRestartPolicyAlways,
-				},
-			},
-			expected: []podresize.ResizableContainerInfo{
-				{
-					Name:      "c1",
-					Resources: &cgroups.ContainerResources{CPUReq: originalCPU, CPULim: originalCPULimit, MemReq: originalMem, MemLim: originalMemLimit},
-				},
-				{
-					Name:          "c1-init",
-					Resources:     &cgroups.ContainerResources{CPUReq: increasedCPU, CPULim: increasedCPULimit, MemReq: increasedMem, MemLim: increasedMemLimit},
-					InitCtr:       true,
-					RestartPolicy: v1.ContainerRestartPolicyAlways,
-				},
-			},
-		},
-		{
-			name: "Burstable QoS pod, one container, one restartable init container - decrease init container CPU only",
-			containers: []podresize.ResizableContainerInfo{
+func doGuaranteedPodResizeTests(f *framework.Framework) {
+	ginkgo.Describe("guaranteed pods with a single regular container", func() {
+		ginkgo.DescribeTableSubtree("with various policies, init containers, and extended resources", func(cpuPolicy, memPolicy v1.ResourceResizeRestartPolicy, hasInitCtrs, resizeInitCtrs, extendedResources bool) {
+			ginkgo.DescribeTable("performing a resize operation", func(ctx context.Context, desiredCPU, desiredMem string) {
```

Comment:
nit: for shorter test names
```suggestion
			ginkgo.DescribeTable("resize", func(ctx context.Context, desiredCPU, desiredMem string) {
```